### ISSUE TITLE
Use PVC Dialog as inspiration for PRC Dialog

### ISSUE THREAD

**Issue by lindseywild** [primer/react#2585]
## Context

It was discovered in the [Dialog v2 draft](https://primer.style/react/drafts/Dialog) guidelines that the Dialog is proposed to stay open when clicking outside. From an accessibility perspective, a dialog should always close when clicking outside.

A lot of work was done on the [PVC Dialog](https://primer.style/view-components/components/alpha/dialog) which has sign-off from the Accessibility team. Can we use the patterns there to inspire the guidelines and behavior for the new PRC Dialog?

A few immediate action items:
* Remove the guidance on the Dialog v2 draft that states "To avoid losing information and missing important messages, clicking outside of the dialog will not close it."
* Remove this behavior from the Dialog and allow clicking outside to close the Dialog

**Comment by lesliecdubs** [comment]
Thanks Lindsey!

@primer/react-reviewers does anyone remember/know who worked on the PRC Dialog v2? I don't recall the team working on this one recently.

**Comment by tallys** [comment]
@langermank can you take on the two immediate action items and then create/link to an issue?

@colebemis has more context - the two people who did the work on it no longer work here.

**Comment by langermank** [comment]
I created two new issues to track this work: 
- https://github.com/github/primer/issues/1550
- https://github.com/github/primer/issues/1551
